examples/adaptive_library/BezierCurveRotorPockets.py

```diff
@@ -0,0 +1,302 @@
+def set_default_parameter(parameter_name, default_value):
+    try:
+        mc.get_adaptive_parameter_value(parameter_name)
+    except pymotorcad.MotorCADError:
+        mc.set_adaptive_parameter_value(parameter_name, default_value)
+
+
+# %%
+# Use the ``set_default_parameter()`` method to set the required ``bez_curve_projection``,
+# ``upperconvex`` and ``lowerconcave`` parameters if undefined.
+set_default_parameter("bez_curve_projection", 6)
+set_default_parameter("upperconvex", 0.5)
+set_default_parameter("lowerconcave", -0.3)
+
+# %%
+# The adaptive parameters will be used to define the curved rotor pocket geometry with a Bezier
+# function. The parameters relate to the rotor pocket shape as follows:
+#
+# * ``bez_curve_projection``  defines the rotor pocket extension beyond the magnet edge in the
+#   direction of the magnet length in mm.
+#
+# * ``upperconvex`` defines the concave rotor pocket extension beyond the magnet edge in the
+#   direction of the magnet thickness. This parameter is dependent on the magnet thickness (a
+#   Standard Template parameter).
+#
+# * ``lowerconcave`` defines the convex rotor pocket curvature in the direction of the magnet
+#   thickness. This parameter is dependent on the magnet thickness (a Standard Template parameter).
+#
+# .. image:: ../../images/Adaptive_Geometry_Bezier_e4a_2.png
+
+
+# %%
+# Get required parameters and objects
+# -----------------------------------
+# From Motor-CAD, get the adaptive parameters and their values.
+#
+# Get the standard template rotor region. This can be drawn for debugging if required.
+rotor_region = mc.get_region("Rotor")
+rotor_radius = mc.get_variable("RotorDiameter") / 2
+poles = mc.get_variable("Pole_Number")
+pole_angle = 360 / (poles * 2)
+
+# %%
+# Get the Adaptive Parameters specified in Motor-CAD, and their values
+totalprojection = mc.get_adaptive_parameter_value("bez_curve_projection")
+upperconvex = mc.get_adaptive_parameter_value("upperconvex")
+lowerconcave = mc.get_adaptive_parameter_value("lowerconcave")
+
+# %%
+# Get the Rotor Pocket regions
+Rotor_Pocket_regions = []
+for i in rotor_region.child_names:
+    if "Pocket" in i:
+        Rotor_Pocket_regions.append(mc.get_region(i))
+
+# %%
+# Get the Magnet Regions
+Magnet_regions = []
+for i in rotor_region.child_names:
+    if "Magnet" in i:
+        Magnet_regions.append(mc.get_region(i))
+
+# %%
+# Find the magnet edge that is shared with the first Rotor Pocket
+for j in Magnet_regions:
+    for i in j.entities:
+        MagnetFaceLine = Rotor_Pocket_regions[0].find_entity_from_coordinates(i.start, i.end)
+        if MagnetFaceLine is not None:
+            break
+
+# %%
+# Get properties of the Magnet edge that will be used to define the new Rotor Pocket geometry
+LineLength = MagnetFaceLine.length
+StartCoordinate = MagnetFaceLine.start
+
+# %%
+# Get properties of the Magnet edge that will be used to define the new Rotor Pocket geometry
+LineLength = MagnetFaceLine.length
```

Comment:
Lines 175-179 seems copied twice. 

Get properties of the Magnet edge that will be used to define the new Rotor Pocket geometry
LineLength = MagnetFaceLine.length
StartCoordinate = MagnetFaceLine.start